src/main/java/com/vmware/vipclient/i18n/messages/api/opt/server/RemoteProductOpt.java

```diff
@@ -82,6 +82,7 @@ public void getSupportedLocales(MessageCacheItem cacheItem) {
                             for (String languageTag : supportedLocales) {
                                 languageTags.put(languageTag, languageTag);
                             }
+                            languageTags.put(ConstantsKeys.SOURCE, ConstantsKeys.SOURCE);
```

Comment:
I know when no local source bundles, 'source' is not in supported locale list, so you have to add it to supported locale list. But it seems strange to supported locale list. 
Is it better change line 114 in ComponentService.java to below:

if (ps.isSupportedLocale(locale) || dto.getLocale().equals(ConstantsKeys.SOURCE) || VIPCfg.getInstance().isPseudo()) {

Reply:
> Why always put "source" into language tags even though no source on service?

There's always source on service. These are the messages from VIP if when you use "latest" as locale. 
We want to use these messages form VIP service for microservices that do not have source locally. e.g. in VCF and Atlas products.

'Right now, we do have the code logic to send use "latest" if locale is source, but it never gets used because getSupportedLocales from service does not include it.

Reply:
> > Why always put "source" into language tags even though no source on service?
> 
> There's always source messages on service. These are the messages from VIP if when you use "latest" as locale. We want to use these messages form VIP service for microservices that do not have source locally. e.g. in VCF and Atlas products.
> 
> Right now, we already have the code logic to use "latest" if locale is source in the request to VIP. However, it never gets used because getSupportedLocales from service does not include "source" in it.

Yes, this works too. I will change it.